### ISSUE TITLE
Add semantic merging for content restructuring

### ISSUE THREAD

**Issue by jakubpawlowicz** [clean-css/clean-css#588]
Depends on #533

**Comment by djibe** [comment]
Hi, [CSSO ](https://github.com/css/csso)merges thoses classes 
```
.foo { color: #ff0000; }
.bar { color: rgba(255, 0, 0, 1); }
```

Into `.bar,.foo{color:red}`

Will CleanCSS integrate this work ?
Thanks

**Comment by jakubpawlowicz** [comment]
Won't fix as clean-css is going into [maintenance mode](https://github.com/clean-css/clean-css/discussions/1209).
